frontend/utility/system-info-macos.mm

```diff
@@ -1,6 +1,96 @@
 #include "system-info.hpp"
 
+#ifdef __aarch64__
+#import <util/platform.h>
+
+#import <Foundation/Foundation.h>
+#import <Foundation/NSProcessInfo.h>
+#import <sys/sysctl.h>
+#import <sys/types.h>
+
+static std::optional<std::string> getCpuName()
```

Comment:
Not a blocker, but iirc it is more idiomatic in C++ to use anonymous namespaces instead of static function/variable declarations (it was only un-deprecated because of pesky C-compatibility but for pure C++ it is considered problematic as `static` has context-specific meanings).